cmd/internal/flowmode/cluster_builder.go

```diff
@@ -62,12 +62,20 @@ func buildClusterService(opts clusterOptions) (*cluster.Service, error) {
 	if config.AdvertiseAddress == "" {
 		advertiseAddress := fmt.Sprintf("%s:%d", net.ParseIP("127.0.0.1"), listenPort)
 		if opts.EnableClustering {
-			addr, err := advertise.FirstAddress(opts.AdvertiseInterfaces)
+			advertiseInterfaces := opts.AdvertiseInterfaces
+			if useAllInterfaces(advertiseInterfaces) {
+				advertiseInterfaces = nil
+			}
+			addr, err := advertise.FirstAddress(advertiseInterfaces)
 			if err != nil {
-				level.Warn(opts.Log).Log("msg", "could not find advertise address using default interface names, "+
+				level.Warn(opts.Log).Log("msg", "could not find advertise address using network interfaces", advertiseInterfaces,
 					"falling back to localhost", "err", err)
```

Comment:
@rfratto @tpaschalis why do we fall back to localhost in this case? As a user, if I specify some advertise interfaces and the agent fails to retrieve a valid IP address from it, I would prefer that it returns an error rather than falling back to localhost, which was not an address I wanted. Also, the user might not directly realize that the IP is wrong because the log in the terminal is drowned in startup logs

Reply:
@thampiotr added this code and would have the most context here.

Reply:
oops sorry, since that it was clustering I just pinged both of you without actually looking at the history

Reply:
I'll look into the overall PR tomorrow, but here's some context for your question. We landed this fix with @thampiotr to fix `main-windows` from when the cluster service was first introduced.

At that point, there was no way to configure the advertise interfaces, so there was no way for the Agent to ever start up (since there was no `eth0` or `en0` by default).

Now that interfaces are configurable we can look into this again, but in any case, we shouldn't fail to start the Agent for something that the user might not care about.

Reply:
Ok makes sense, thanks.
I still prefer a missconfigured program to crash on startup rather than falling back on an alternative solution, especially that fixing the problem is easy now. But you made a point that some users might not care and be fine with the fall back. If you believe that the fall back is the best option then let's keep it

Reply:
I'm still of the opinion that falling back to localhost is the correct option here.

It's unlikely that someone would deploy a cluster on their _first_ interaction with the Agent, and even if they did, they'd read the docs about the flags and the default interfaces on Windows.

In general, we shouldn't surprise people by having the Agent fail and having to refer them to the docs so they can learn about a parameter they don't even need at that time.